secboot/signme.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+log() {
+    echo "[$(date +'%Y-%m-%d %H:%M:%S')] $*"
+}
+
+err_report() {
+    log "[!] Error on line $1"
+    exit 1
+}
+trap 'err_report $LINENO' ERR
+
+# Input and constants
+if [[ $# -ne 1 ]]; then
+    log "[!] Usage: $0 <disk-image.zst>"
+    exit 1
+fi
+
+DISK_IMAGE_ZST="$1"
+DISK_IMAGE="disk.raw"
+EFI_IMAGE="efi-partition.img"
+SIGNED_EFI="BOOTX64.EFI.signed"
+CERT="uefi-signing-cert.pem"
+KEY="vault:ghaf-secureboot-testkv:uefi-signing-key"
+REPO="harbor.ppclabz.net/ghaf-secboot/ghaf-uefi"
+TAG="signed"
+ZSTD_IMAGE="ghaf_0.0.1.raw.zst"
+
+case "$1" in
+    *.iso)
+	input_type="iso"
+	log "ISO Image detected"
+	;;
+    *.zst)
+	input_type="zst"
+	log "ZST'ed Image detected"
+	;;
+    *)
+	log "Unsupported input file: $1" >&2
+	exit 1
+	;;
+esac
+
+# Ensure required tools are available
+for cmd in zstd fdisk dd mcopy sbsign oras; do
+    if ! command -v "$cmd" &>/dev/null; then
+        log "[!] Required tool '$cmd' not found in PATH"
+        exit 1
+    fi
+done
+
+# Jenkins global credentials / Jenkins secrets are used for now
```

Comment:
Can you add a top-level comment explaining what environment vars this assumes to be set? Is it only ORAS_{USERNAME,PASSWORD}?